File: 2_Signale/Python/MSMP_Sprungfunktion.py
import numpy as np
import matplotlib.pyplot as plt

def sprung_heaviside(x, M):
    return np.heaviside(x-M, 1)

def sprung_signum(x, M):
    y = 2/(1+np.sign(x-M))
    index = np.argwhere(x==M)
    y[index] = 1
    return y

def sprung_where(x, M):
    y = np.zeros(x.shape)
    indices = np.argwhere(x<M)
    y[indices] = 0
    indices = np.argwhere(x>=M)
    y[indices] = 1
    return y
    
def sprung_schleife(x, M):
    y = np.ones(x.shape)
    for idx in range(0, len(x)):
        if x[idx]<M:
            y[idx] = 0
    return y

def sprung_indexing(x, M):
    y = np.ones(x.shape)
    y[x<M] = 0
    return y

x = np.linspace(-2, 8, 11)
M = 3
y = sprung_indexing(x, 4)

plt.stem(x, y)

    

File: 2_Signale/MATLAB/MSMP_Sprungfunktion.m
x = 1:0.5:8;
M = 2;
y = sprung_schleife(x, 4);
stem(x, y)

function y = sprung_heaviside(x, M)
    y = heaviside(x-M);
    y(x==M) = 1;
end

function y = sprung_sign(x, M)
    y = sign(x-M);
    index = find(x==M);
    disp(index)
    y(x==M)=1;
    y = (1.+y)/2;
end

function y = sprung_indexing(x, M)
    y = ones(size(x));
    y(x<M)=0;
end

function y = sprung_schleife(x, M)
    y = ones(1,length(x));
    for idx=1:length(x)
       if x(idx)<M
           y(idx)=0;
       end
    end
end